Unable to load the config property type: [Ljava.lang.String through native image

The application is running in error when launching a native image of an application getting a config property as a String[] : 

```
./target/services-matching-quarkus-1.0.0-SNAPSHOT-runner 
java.lang.IllegalStateException: Unable to load the config property type: [Ljava.lang.String;
	at io.quarkus.arc.runtime.ConfigRecorder.load(ConfigRecorder.java:70)
	at io.quarkus.arc.runtime.ConfigRecorder.validateConfigProperties(ConfigRecorder.java:29)
	at io.quarkus.deployment.steps.ConfigBuildStep$validateConfigProperties24.deploy_0(ConfigBuildStep$validateConfigProperties24.zig:188)
	at io.quarkus.deployment.steps.ConfigBuildStep$validateConfigProperties24.deploy(ConfigBuildStep$validateConfigProperties24.zig:36)
	at io.quarkus.runner.ApplicationImpl1.doStart(ApplicationImpl1.zig:210)
	at io.quarkus.runtime.Application.start(Application.java:94)
	at io.quarkus.runtime.Application.run(Application.java:218)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:41)
2019-11-07 08:21:06,531 WARN  [io.qua.arc] (main) An error occured during delivery of the @BeforeDestroyed(ApplicationScoped.class) event: java.lang.NullPointerException
	at io.quarkus.mongodb.runtime.MongoClientProducer.onStop(MongoClientProducer.java:20)
	at io.quarkus.mongodb.runtime.MongoClientProducer_Observer_onStop_6e82549763245071a3987b5e25ba17616004a313.notify(MongoClientProducer_Observer_onStop_6e82549763245071a3987b5e25ba17616004a313.zig:53)
	at io.quarkus.arc.EventImpl$Notifier.notify(EventImpl.java:228)
	at io.quarkus.arc.EventImpl$Notifier.notify(EventImpl.java:218)
	at io.quarkus.arc.ArcContainerImpl.shutdown(ArcContainerImpl.java:275)
	at io.quarkus.arc.Arc.shutdown(Arc.java:37)
	at io.quarkus.arc.runtime.ArcRecorder$1.run(ArcRecorder.java:37)
	at io.quarkus.runtime.StartupContext.close(StartupContext.java:43)
	at io.quarkus.runner.ApplicationImpl1.doStart(ApplicationImpl1.zig:276)
	at io.quarkus.runtime.Application.start(Application.java:94)
	at io.quarkus.runtime.Application.run(Application.java:218)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:41)

2019-11-07 08:21:06,535 INFO  [io.sma.rea.mes.ext.MediatorManager] (main) Cancel subscriptions
Exception in thread "main" java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl1.doStart(ApplicationImpl1.zig:282)
	at io.quarkus.runtime.Application.start(Application.java:94)
	at io.quarkus.runtime.Application.run(Application.java:218)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:41)
Caused by: java.lang.IllegalStateException: Unable to load the config property type: [Ljava.lang.String;
	at io.quarkus.arc.runtime.ConfigRecorder.load(ConfigRecorder.java:70)
	at io.quarkus.arc.runtime.ConfigRecorder.validateConfigProperties(ConfigRecorder.java:29)
	at io.quarkus.deployment.steps.ConfigBuildStep$validateConfigProperties24.deploy_0(ConfigBuildStep$validateConfigProperties24.zig:188)
	at io.quarkus.deployment.steps.ConfigBuildStep$validateConfigProperties24.deploy(ConfigBuildStep$validateConfigProperties24.zig:36)
	at io.quarkus.runner.ApplicationImpl1.doStart(ApplicationImpl1.zig:210)
	... 3 more
```

This bug is not reproduced when launching the application without the native image with `mvn clean compile quarkus:dev`.

application.properties : 

```partners.available=ABC,DEF```

PartnerService.java : 

```
package org.acme.quickstart;

import org.eclipse.microprofile.config.inject.ConfigProperty;

import javax.enterprise.context.ApplicationScoped;
import java.util.stream.Stream;

@ApplicationScoped
public class PartnerService {

    @ConfigProperty(name="partners.available")
    private String[] availablePartners;

    public Stream<String> findByService() {
        return Stream.of(availablePartners);
    }
}

```

Updating the availablePartners from String[] to List<String> solve the problem.

Any idea why it is not working through the native image running?

**Steps to reproduce the behavior:**
1. Set a config property to a list of String such as "partners.available=ABC,DEF"
2. Get it from @ConfigProperty into a String[].
3. Build the native image with such a command : `docker run -v $PWD:/sources --workdir=/sources -v $HOME/.m2/repository:/home/quarkus/.m2/repository -it centos-quarkus-maven:19.2.1 mvn -Pnative package
`
4. Launch the native image with : `./target/application.runner`

**Environment :**
 - Output of `uname -r`:  5.0.0-32-generic
 - Output of `java -version`: openjdk version "1.8.0_222"
 - GraalVM version (if different from Java): 19.2.1
 - Quarkus version or git rev: 0.25.0

**Additional context**
(Add any other context about the problem here.)

--- a/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java
+++ b/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java
@@ -131,7 +131,7 @@ public class ConfigBuildStep {
         for (ConfigPropertyBuildItem item : configProperties) {
             Type requiredType = item.getPropertyType();
             String propertyType = requiredType.name().toString();
-            if (requiredType.kind() != Kind.ARRAY && requiredType.kind() != Kind.PRIMITIVE) {
+            if (requiredType.kind() != Kind.PRIMITIVE) {
                 reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, propertyType));
             }
         }

--- a/integration-tests/main/src/main/java/io/quarkus/it/rest/ServiceWithConfig.java
+++ b/integration-tests/main/src/main/java/io/quarkus/it/rest/ServiceWithConfig.java
@@ -13,11 +13,18 @@ public class ServiceWithConfig {
     @ConfigProperty(name = "web-message")
     String message;
 
+    @ConfigProperty(name = "names")
+    String[] names;
+
     public String host() {
         return quarkusHost;
     }
 
     public String message() {
         return message;
+    }
+
+    public String[] names() {
+        return names;
     }
 }

--- a/integration-tests/main/src/main/java/io/quarkus/it/rest/TestResource.java
+++ b/integration-tests/main/src/main/java/io/quarkus/it/rest/TestResource.java
@@ -78,6 +78,13 @@ public class TestResource {
     }
 
     @GET
+    @Path("/config/names")
+    @Produces("application/json")
+    public String configNames() {
+        return String.join(",", config.names());
+    }
+
+    @GET
     @Path("/count")
     public int count() {
         return count.incrementAndGet();

--- a/integration-tests/main/src/main/resources/application.properties
+++ b/integration-tests/main/src/main/resources/application.properties
@@ -22,6 +22,7 @@ quarkus.resteasy.gzip.enabled=true
 quarkus.resteasy.gzip.max-input=10
 quarkus.http.limits.max-body-size=1K
 web-message=A message
+names=quarkus,redhat
 schedulerservice.cron.expr=0/10 * * * * ?
 
 microprofile.custom.value = 456

--- a/integration-tests/main/src/test/java/io/quarkus/it/main/JaxRSTestCase.java
+++ b/integration-tests/main/src/test/java/io/quarkus/it/main/JaxRSTestCase.java
@@ -37,6 +37,11 @@ public class JaxRSTestCase {
     }
 
     @Test
+    public void testConfigInjectionOfStringArray() {
+        RestAssured.when().get("/test/config/names").then().body(is("quarkus,redhat"));
+    }
+
+    @Test
     public void testAnnotatedInterface() {
         RestAssured.when().get("/interface").then().body(is("interface endpoint"));
 

